### PR TITLE
PLTFRS-8704 Beta Feature Management

### PR DESCRIPTION
**Pull request by frntz**
https://issues.sierrawireless.com/browse/PLTFRS-8704

#### Changes proposed in this pull-request
* Allow query on a dynamoDB table
* Update and Put mixed item without the need to wrap all value in JSON

#### Version bump
minor

### CODE REVIEW

File: techDynDb.js [hunk 1]
```diff
@@ -142,6 +187,40 @@ module.exports = function(configuration) {
...
+    var dynData = dynTypes.AttributeValue.wrap(item ? getDynData(item, jsonKeys) : {});
+
+    return new BPromise(function(resolve, reject) {
+      dyn.putItem(
+        {
+          Item: _.assignIn(dynKey, dynData),
+          TableName: tableName
+        },
+        function(err, data) {
+          notify('ruuid', key, 'Create', techTime.end(start));
```
Comment (jlescot) [inline techDynDb.js]: 'Create' > 'Update'?

File: techDynDb.js [hunk 2]
```diff
@@ -113,6 +112,52 @@ module.exports = function(configuration) {
...
+   * @return {BPromise}
+   */
+  function wrapUpdateMixedItem(ruuid, tableName, key, newValues, jsonKeys) {
+    var start = techTime.start();
+
+    var updateParams = {
+      Key: dynTypes.AttributeValue.wrap(key),
+      AttributeUpdates: dynTypes.AttributeValueUpdate.put(getDynData(newValues, jsonKeys)),
+      TableName: tableName,
+      ReturnValues: 'ALL_NEW'
```
Comment (jlescot) [inline techDynDb.js]: Seems that the returned values are simply ignored afterwards... 🤔

File: techDynDb.js [hunk 3]
```diff
@@ -113,6 +112,52 @@ module.exports = function(configuration) {
     });
   }
 
+  //
+  /**
+   * Returns an object compatible with dynamoDB
+   *
```
Comment (fskandrani) [inline techDynDb.js]: :gun:

File: techDynDb.js [hunk 4]
```diff
@@ -113,6 +112,52 @@ module.exports = function(configuration) {
     });
   }
 
+  //
```
Comment (fskandrani) [inline techDynDb.js]: :gun:

**Pull request merged**